CARDS-2424 and CARDS-2402: New Phone Call Follow-Up and replacement Event - Clinical Cardiac forms

acrowthe opened the pull request.

acrowthe committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form

acrowthe committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form
Fix missing validation error text

acrowthe committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form
Update forms from demo feedback

acrowthe committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form
Add option for other details

acrowthe committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form
Add option for other details to phone call form

Hajany commented:
Remove highlighted text - this is automatically done in the backend, don't need to show on UI
![image](https://github.com/data-team-uhn/cards/assets/96542282/735ae598-0546-4271-b8ff-7bc52dcf808c)

Add event type abbreviation [ex. (HF), (VHD), (CV Other)] to the Subtype question - match formatting for "Coronary Artery Disease (CAD) Subtype"
![image](https://github.com/data-team-uhn/cards/assets/96542282/70506bb6-d01b-48fa-9fd5-322396e34430)
![image](https://github.com/data-team-uhn/cards/assets/96542282/24fe1207-d1fb-4e3a-b513-42dc589180d9)

Add "the" it should read: "Why did the patient not provide consent?"
![image](https://github.com/data-team-uhn/cards/assets/96542282/89fdd95d-1d1a-4050-9660-24c4b1157ff5)

This question should be "How long was the hospital stay" not "How many days ago" this can be interpreted differently so let's stick to the specs. Also, can we add a warning if [below 1 or above 365] on the spot similar to questions in the Admin section? Right now I can add anything outside this range without a warning and save - it only show "invalid" after saving
![image](https://github.com/data-team-uhn/cards/assets/96542282/3ea25afe-76d7-416b-b582-62d93449d0e6)

Add text "Diagnostic Test Date" to question & remove "event date relative to enrollment or study start" as this is automatically done in the backend
![image](https://github.com/data-team-uhn/cards/assets/96542282/2b110f5f-6f0f-4b4a-9083-b3ac08987f81)

Missing an answer option of "**Troponin normal**" under CV Other & Afib should be formatted to AFib
![image](https://github.com/data-team-uhn/cards/assets/96542282/5f473769-7171-4bcd-9af2-4bcfb993430d)

Remove highlighted text - this is automatically done in the backend, don't need to show on UI
![image](https://github.com/data-team-uhn/cards/assets/96542282/c191a520-82b6-49ac-88ec-d52d6191e13d)

Missing answer options of:

- Volume overload improved
- Volume overload unchanged
- Volume overload worsened
![image](https://github.com/data-team-uhn/cards/assets/96542282/f6b8adac-665a-4ee5-84c4-a8aa7e2a3dbe)

acrowthe committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form
Further form updates

marta- committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form
Updated condition to mask conditional section values bug

marta- committed:
CARDS-2424: Phone Call Follow-Up
Updated the computed list of conditions ecpression to match the new syntax for optional value

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Event-ClinicalCardiac.xml:
```diff
@@ -0,0 +1,10150 @@
+				</property>
+				<property>
+					<name>expression</name>
+					<value>return @{[ec_37_other]}.join(", ")</value>
```
Is this really what they want? That is a multivalued answer, and it is being treated like a single procedure. I would expect one new section for each answer.

![image](https://github.com/data-team-uhn/cards/assets/88663/a8b6ea7c-3a53-4ca0-8f7c-1d73f7dfc260)

marta- replied:
We were told they would almost never need to add more than one extra procedure, and are happy with just one meta section.

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Event-ClinicalCardiac.xml:
```diff
@@ -0,0 +1,10150 @@
+			</property>
+			<property>
+				<name>text</name>
+				<value>Start date from</value>
```
What does this mean?

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Event-ClinicalCardiac.xml:
```diff
@@ -0,0 +1,10150 @@
+				</property>
+				<property>
+					<name>text</name>
+					<value>Procedure coding system/ source</value>
```
```suggestion
					<value>Procedure coding system/source</value>
```

This needs fixing in many other places.

But this isn't clear to me, is it just the name of the system in use, e.g. `ICD-10`, or an actual item like `V97.33XD`?

marta- replied:
Please commit this change before merging.

It's the name of the system. The users will know what to enter.

sdumitriu replied:
Then why not make it a radio/checkbox?

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Event-ClinicalCardiac.xml:
```diff
@@ -0,0 +1,10150 @@
+				</property>
+				<property>
+					<name>text</name>
+					<value>What other procedure not specified in the list was performed?</value>
```
Since this is a multivalued answer, shouldn't this be "What other procedureS ... were performed"?

marta- replied:
Only one is expected and recommended to be entered. The answer is multivalued in the unlikely case there's actually more than one

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/PhoneCallFollow-Up.xml:
So this whole form is supposed to only be used when the whole conversation is done?

marta- replied:
I'm not following.

sdumitriu replied:
I was quite sure that the `Date closed` question is mandatory and a precondition for enabling the second page. Turns out that's not the case, only the `Date opened` question behaves that way.

marta- committed:
CARDS-2424: Phone Call Follow-Up
Updated some labels, made symptom list required

acrowthe committed:
CARDS-2424: Phone Call Follow-Up
Set minAnswers to 1 for pc_12_3_6

sdumitriu committed:
CARDS-2402: Heracles: new Event - Clinical Cardiac form to replace existing form
Minor whitespace change

sashaandjic reviewed (approved):
tested

sdumitriu committed:
CARDS-2424, CARDS-2402: Phone Call Follow-Up, Replacment Event Clinical Cardiac form
Add spaces in the file names

sdumitriu committed:
CARDS-2402: Heracles: new Event - Clinical Cardiac form to replace existing form
Minor white space change, again

sashaandjic reviewed (approved):
tested
![image](https://github.com/data-team-uhn/cards/assets/23243244/3aaee309-e4dc-4069-8e45-74a17000a823)

![image](https://github.com/data-team-uhn/cards/assets/23243244/b04b549f-38d2-4397-b20d-7cec164ceb55)

sashaandjic reviewed (approved):
re-tested
